### ISSUE TITLE
Bundler adding extra complication to externally referenced schemas

### ISSUE THREAD

**Issue by steve-nay-sage** [Redocly/redocly-cli#818]
**Describe the bug**
This is a variation of issue #349 that you fixed for me several months ago. 

If you have something like this in components.schemas:

```
components:
  schemas:
    myObject:
      $ref: ../path/to/myObject.v2.schema.yaml
```

The bundler output will look something like this:

```
components:
  schemas:
    myObject:
      $ref: '#/components/schemas/myObject.v2.schema'
    myOjbect.v2.schema:
      title: some object
      type: object
      description: This is the example object
      properties:
        key:
          type: string
          description: UUID
          readOnly: true
          .
          .
          .
```

While the extra schema is in some ways invisible (that is, using myObject still ultimately gets to the right schema), adding the schema that has a name based on the file name is completely unnecessary. It would be kind of embarrassing to give this OpenAPI definition to someone who wants to build a client for my service because they'd ask about the unneeded indirection.

**Expected behavior**

I would expect the output to look like this:

```
components:
  schemas:
    myObject:
      title: some object
      type: object
      description: This is the example object
      properties:
        key:
          type: string
          description: UUID
          readOnly: true
          .
          .
          .
```

There is no need to create a new schema; just replace the $ref line with the contents from the referenced file. 

**OpenAPI definition**
OAS v3.0

**Redocly Version(s)**
$ redocly --version
1.0.0-beta.102

**`Node.js` Version(s)**
v18.7.0

**Comment by tatomyr** [comment]
Hi @steve-nay-sage,

Thanks for creating the issue. How would you expect it to behave if we have several entries of the same external ref in the `components`? For instance:

```yaml
components:
  schemas:
    myObject:
      $ref: '#/components/schemas/myObject.v2.schema'
    meAnotherObject: 
      $ref: '#/components/schemas/myObject.v2.schema'
```

**Comment by steve-nay-sage** [comment]
That's a good question @tatomyr. I can see this scenario for error message schemas where you'd want to use the same schema for 400 errors, 404 errors, etc.

Without thinking this through deeply, I would expect this to work just like what I described above. That is, replace the external references with the referenced schema. The schema would appear twice (or more) in the resulting file, but with different object names each time. I would say that's expected behavior and not a problem. The current behavior only brings the external schema in once, but I don't think that's dramatically more efficient than copying it in at every place that it's referenced in components.schemas. 

From my perspective, the redirection and unexpected objects with odd names are more confusing. In your example, you are using camelCase object names for myOjbect and meAnotherOjbect, but the bundle operation would result in a new object named myObject.v2.schema, which is not camel case and thus breaks the object naming model. My proposal avoids that.

**Comment by tatomyr** [comment]
Thanks for you thoughts @steve-nay-sage,

That totally makes sense to me. We checked it and it produces an extra field in `components` even with `--dereferenced` option. I believe we can even try resolving the first reference in components and then reuse it.

**Comment by RomanHotsiy** [comment]
Just a note: this is related to https://github.com/Redocly/redocly-cli/issues/290 in some way.

**Comment by tatomyr** [comment]
Should be fixed in `v1.0.0-beta.127`.
